DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
+                "$($currentSettings['PrefixLength'])."
+                ) -join "" | Out-String
+            )
+            $requiresChanges = $true
+        }
+        else
+        {
+            Write-Verbose -Message "Subnet mask is correct."
+        }
 
-            #Apply is true in the case of set - target resource - in which case, it will set the new IP Address
+        #Test if the Default Gateway passed is equal to the current default gateway
+        if($DefaultGateway)
+        {
+            $gateway = $DefaultGateway
+            if(!([System.Net.Ipaddress]::TryParse($gateway, [ref]0)))
+            {
+                throw ( @(
+                    "Default Gateway *$DefaultGateway* is NOT in the correct format. Please "
+                    "correct the gateway ipaddress in the configuration and try again"
+                    ) -join "" | Out-String
+                )
+            }
+
+            $netIpConf = Get-NetIPConfiguration -InterfaceAlias $InterfaceAlias -ErrorAction Stop
+            $currentSettings['DefaultGateway'] = (
+                $netIpConf."$($AddressFamily)DefaultGateway"
+                ).NextHop
+
+            if(!$currentSettings['DefaultGateway'].Equals($DefaultGateway))
+            {
+                Write-Verbose -Message ( @(
+                    "Default gateway does NOT match desired state. Expected $DefaultGateway, "
+                    "actual $($currentSettings['DefaultGateway'])."
+                    ) -join "" | Out-String
+                )
+                $requiresChanges = $true
+            }
+            else
+            {
+                Write-Verbose -Message "Default gateway is correct."
+            }
+        }
+
+        #Test if DHCP is already disabled
+        if(!(Get-NetIPInterface -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily).Dhcp.ToString().Equals('Disabled'))
+        {
+            Write-Verbose -Message "DHCP is NOT disabled."
+            $requiresChanges = $true
+        }
+        else
+        {
+            Write-Verbose -Message "DHCP is already disabled."
+        }
+
+        if($requiresChanges)
+        {
+            # Apply is true in the case of set - target resource - in which case, it will apply the
+            # required IP configuration
             if($Apply)
             {
-                Write-Verbose -Message "Setting IPAddress ..."
+                Write-Verbose -Message ( @(
+                    "At least one setting differs from the passed parameters. Applying "
+                    "configuration..."
+                    ) -join "" | Out-String
+                )
+
+                # Build parameter hash table
+                $Parameters = @{}
                 $Parameters["IPAddress"] = $IPAddress
                 $Parameters["PrefixLength"] = $SubnetMask
                 $Parameters["InterfaceAlias"] = $currentIP[0].InterfaceAlias
```

Comment:
So here we are taking the first item in the $CurrentIP array where on line 151 we take all of them. We should probably try to address that.